cirq/linalg/decompositions.py

```diff
@@ -854,3 +857,145 @@ def kak_decomposition(unitary_object: Union[np.ndarray, 'cirq.SupportsUnitary'],
         global_phase=g * inner_cannon.global_phase,
         single_qubit_operations_before=(b1, b0),
         single_qubit_operations_after=(a1, a0))
+
+
+def kak_vector(unitary: Union[Iterable[np.ndarray], np.ndarray],
+               *,
+               rtol: float = 1e-5,
+               atol: float = 1e-8,
+               check_preconditions: bool = True) -> np.ndarray:
+    r"""Compute the KAK vectors of one or more two qubit unitaries.
+
+    Any 2 qubit unitary may be expressed as
+
+    $$ U = k_l A k_r $$
+    where $k_l, k_r$ are single qubit (local) unitaries and
+
+    $$ A= \exp \left(i \sum_{s=x,y,z} k_s \sigma_{s}^{(0)} \sigma_{s}^{(1)}
+                 \right) $$
+
+    The vector entries are ordered such that
+        $$ 0 ≤ |k_z| ≤ k_y ≤ k_x ≤ π/4 $$
+    if $k_x$ = π/4, $k_z \geq 0$.
+
+    Args:
+        unitary: A unitary matrix, or a multi-dimensional array of unitary
+            matrices. Must have shape (..., 4, 4), where the last two axes are
+            for the unitary matrix and other axes are for broadcasting the kak
+            vector computation.
+        rtol: Per-matrix-entry relative tolerance on equality. Used in unitarity
+            check of input.
+        atol: Per-matrix-entry absolute tolerance on equality. Used in unitarity
+            check of input. This also determines how close $k_x$ must be to π/4
+            to guarantee $k_z$ ≥ 0. Must be non-negative.
+        check_preconditions: If set, verifies that the input corresponds to a
```

Comment:
The `if set` is confusing here. It reads like if you don't pass the value it won't verify.

`When set to False, skips verifying that the input is unitary in order to increase performance.`

Reply:
I'll modify the docstring then merge the commit.